### PR TITLE
Remove Decoder type and just use Binder instead.

### PR DESCRIPTION
**Pull request by Choc13**

### CODE REVIEW

File: tests/Symbolica.Extensions.Configuration.FSharp.Tests/BindTests.fs [hunk 1]
```diff
@@ -144,57 +218,171 @@ module OptValue =
...
+    let ``should be Success value if can be converted to bool`` value =
+        test <@ value |> string |> Binder.run Bind.bool = Success(value) @>
 
+    [<Fact>]
+    let ``should be Failure if string can not be converted to bool`` =
         test
-            <@ missingKey
-               |> Bind.optValueOf Decode.int
-               |> Binder.eval section = Success(None) @>
+            <@ "string" |> Binder.run Bind.bool = Failure([ $"Could not decode 'string' as type 'System.Boolean'." ]) @>
```
Comment (iscottb122) [inline tests/Symbolica.Extensions.Configuration.FSharp.Tests/BindTests.fs:267]: The `$` here is not actually needed since the string is not interpolated. You may or may not care. This is the first of 11 such cases, all in this file.

File: tests/Symbolica.Extensions.Configuration.FSharp.Tests/BinderTests.fs [hunk 2]
```diff
@@ -15,6 +13,16 @@ module Result =
     let ``should create a Binder that ignoes the config and returns Success`` (x: int) (config: string) =
         test <@ x |> Binder.result |> Binder.eval config = Success(x) @>
 
+module Fail =
+    [<Property>]
+    let ``should create a Binder that ignoes the config and returns Failure`` (e: string) (config: string) =
```
Comment (iscottb122) [inline tests/Symbolica.Extensions.Configuration.FSharp.Tests/BinderTests.fs:18]: You have an "ignoes" here and also just above. It's not part of this PR but you also have an "evaulates" in `Binder.fs`.

**Review by iscottb122 — APPROVED** [review]

File: tests/Symbolica.Extensions.Configuration.FSharp.Tests/BinderTests.fs (hunk 2 above)
Comment (Choc13) [reply tests/Symbolica.Extensions.Configuration.FSharp.Tests/BinderTests.fs:18]: Yeah I spotted the evaluates one and fixed in a later PR. I'll do ignoes now though.

Comment (Choc13) [reply tests/Symbolica.Extensions.Configuration.FSharp.Tests/BinderTests.fs:18]: Will fix these in the next PR.

File: tests/Symbolica.Extensions.Configuration.FSharp.Tests/BindTests.fs (hunk 1 above)
Comment (Choc13) [reply tests/Symbolica.Extensions.Configuration.FSharp.Tests/BindTests.fs:267]: Will fix in the next PR.
